tkltest/util/coverage_util.py

```diff
@@ -44,28 +46,88 @@ def get_coverage_for_test_suite(build_file, build_type, test_root_dir, report_di
     if build_type == 'ant':
-        command_util.run_command("ant -f {} merge-coverage-report".format(build_file), verbose=False)
+        cmd = "ant -f {} merge-coverage-report".format(build_file)
         jacoco_raw_date_file = os.path.join(test_root_dir, "merged_jacoco.exec")
     elif build_type == 'maven':
-        command_util.run_command("mvn -f {} clean verify site".format(build_file), verbose=False)
+        cmd = "mvn -f {} clean verify site".format(build_file)
         # in case of maven only, jacoco.exec is created inside the monolithic subdir of the cud-amplified tests
         if os.path.isdir(os.path.join(test_root_dir, "monolithic")):
             jacoco_raw_date_file = os.path.join(test_root_dir, "monolithic", "jacoco.exec")
         else:
             jacoco_raw_date_file = os.path.join(test_root_dir, "jacoco.exec")
     else: #gradle
-        command_util.run_command("gradle --project-dir {} tklest_task".format(test_root_dir), verbose=False)
+        cmd = "gradle --project-dir {} tklest_task".format(test_root_dir)
+        jacoco_raw_date_file = os.path.join(test_root_dir, "jacoco.exec")
+    try:
+        command_util.run_command(cmd, verbose=False)
+    except subprocess.CalledProcessError as e:
+        tkltest_status('Error while running test suite for coverage computing: {}\n{}'.format(e, e.stderr), error=True)
+        sys.exit(1)
+    if not os.path.exists(jacoco_raw_date_file):
+        tkltest_status('Error: {} was not created by : {}'.format(jacoco_raw_date_file, cmd), error=True)
+        sys.exit(1)
+
+    if additional_test_suite:
+        '''
+        when we have additional_test_suite, we allow it to fail.
+        in case of failure, we will use the original .csv & .exec files
+        as long as no_failure flag is true, we continue with the code 
+        '''
+        no_failure = True
+        additional_build_targets = ' '.join(additional_test_suite['build_targets'])
+        additional_build_file = additional_test_suite['build_file']
+        dev_build_type = additional_test_suite['build_type']
+        if dev_build_type == 'ant':
+            cmd = "ant -f {} {}".format(additional_build_file, additional_build_targets)
+        elif dev_build_type == 'maven':
+            cmd = "mvn -f {} {}".format(additional_build_file, additional_build_targets)
+        else:  # gradle
+            cmd = "gradle --project-dir {} {}".format(os.path.dirname(additional_build_file), additional_build_targets)
+        try:
+            command_util.run_command(cmd, verbose=False)
+        except subprocess.CalledProcessError as e:
+            tkltest_status('Error while running test suite for coverage computing:\n {}\n{}'.format(e, e.stderr))
+            no_failure = False
+        additional_exec_file = additional_test_suite['coverage_exec_file']
+        if no_failure and not os.path.exists(additional_exec_file):
+            tkltest_status('Error: {} was not created using command: {}'.format(additional_exec_file, cmd))
```

Comment:
@haim-kermany: For printing error messages, the preferred way is to set the option `error=True` in the call to `tkltest_status`. This option causes `ERROR: `to be prepended to the message; the message parameter need not contain the `Error:` prefix. See https://github.com/konveyor/tackle-test-generator-cli/blob/main/tkltest/util/logging_util.py#L36-L40